spec/soft-tabs-layer-spec.coffee

```diff
@@ -0,0 +1,57 @@
+TextBuffer = require "../src/text-buffer"
+Point = require "../src/point"
+
+describe "soft tabs layer", ->
+  [text, softTabsLayer, linesLayer] = []
+
+  beforeEach ->
+    text = "    hello  world ;)"
```

Comment:
Might be nice to test the interaction when there are newlines and hard tabs in the underlying buffer. Maybe not for every spec, but just have an additional spec that constructs a `TextBuffer` with content to test this case.